Sources/Comet/Authenticator/Authenticator.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  Authenticator.swift
+//  Comet
+//
+//  Created by Tuan Tu Do on 26.05.2021.
+//  Copyright © 2021 Etnetera. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+/// TODO
+final class Authenticator {
+    private let tokenProvider: TokenProviding
+    private let queue = DispatchQueue(label: "Authenticator-\(UUID().uuidString)")
+    private var refreshTokenPublisher: AnyPublisher<String, AuthenticatorError>?
+
+    init(tokenProvider: TokenProviding) {
+        self.tokenProvider = tokenProvider
+    }
+
+    /// TODO
+    var accessToken: AnyPublisher<String, AuthenticatorError> {
```

Comment:
Wouldnt be better `Future` instead of `AnyPublisher`? Same for refresh... 🤔

Reply:
`Future` is a `Publisher` as wel...